app/helpers/checks.py

```diff
@@ -50,38 +46,9 @@ def check_params(scheme, host, url, base_path):
             f"The url given as parameter was too long. (limit is 2046 "
             f"characters, {len(url)} given)"
         )
-    hostname = urlparse(url).hostname
-    if hostname is None:
-        logger.error("Could not determine hostname from the following url : %s", url)
-        abort(400, 'Could not determine the query hostname')
-    domain = ".".join(hostname.split(".")[-2:])
-    if domain not in allowed_domains and hostname not in allowed_hosts:
-        logger.error(
-            "neither the hostname (%s) nor the domain(%s) are part of their "
-            "respective allowed list of domains (%s) or "
-            "hostnames(%s)",
-            hostname,
-            domain,
-            ', '.join(allowed_domains),
-            ', '.join(allowed_hosts)
-        )
-        abort(400, 'Neither Host nor Domain in the url parameter are valid')
-    if host not in allowed_hosts:
-        """
-        This allows for compatibility with dev hosts or local builds for testing purpose.
-        """
-        host = host.replace(
-            '://', ''
-        )  # We make sure here that the :// can't get duplicated in the shorturl
-        base_url = ''.join((scheme, '://', host, base_path if 'localhost' not in host else ''))
-        base_url = ''.join((
-            base_url,
-            'v4/shortlink/shortlinks/' if base_url.endswith('/') else '/v4/shortlink/shortlinks/'
-        ))
-    else:
-        base_url = ''.join((scheme, '://s.geo.admin.ch/'))
-
-    return base_url
+    if not re.match(ALLOWED_DOMAINS_PATTERN, urlparse(url).hostname):
```

Comment:
Note sure if this is already a use case tested by unittest, but I guess think here we need to make sure that an  invalid `url` (e.g. `url=guguseli`) don't crash and return a HTTP 400.

Reply:
Think that has been implemented before. When condensing stuff here, it got lost

Reply:
Implemented using the `validators` package.